acceptance_tests/features/steps/rh_ui_cookies.py

```diff
@@ -0,0 +1,110 @@
+    context.browser.find_by_id("ons-cookies-banner")
+
+
+@step("the user accepts the cookies on the cookies banner")
+def accept_cookies_on_cookies_banner(context):
+    xpath_string = '//button[contains(@class,"ons-js-accept-cookies")]'
+    context.browser.find_by_xpath(xpath_string).click()
+
+
+@step("the user rejects the cookies on the cookies banner")
+def reject_cookies_on_cookies_banner(context):
+    xpath_string = '//button[contains(@class,"ons-js-reject-cookies")]'
+    context.browser.find_by_xpath(xpath_string).click()
+
+
+@step("the 'View cookies' hyperlink points to {expected_path}")
+def click_view_cookies_link_on_banner(context, expected_path):
+    xpath_string = '//a[@class="ons-cookies-banner__link" and text()="View cookies"]'
+    link = context.browser.find_by_xpath(xpath_string).first
+    test_helper.assertEqual(link["href"], f"{Config.RH_UI_URL}{expected_path}")
+
+
+@step("the 'cookies' hyperlink on the cookies banner points to {expected_path}")
+def click_cookies_hyperlinked_text_on_banner(context, expected_path):
+    xpath_string = '//div[@class="ons-cookies-banner__statement"]/a[text()=" cookies."]'
+    link = context.browser.find_by_xpath(xpath_string)
+    test_helper.assertEqual(link["href"], f"{Config.RH_UI_URL}{expected_path}")
+
+
+@step('the "change cookie preferences" hyperlink text points to {expected_path}')
+def click_change_cookie_preferences_link_on_banner(context, expected_path):
+    xpath_string = '//span[@class="ons-cookies-banner__preferences-text"]/a[text()=" change your cookie preferences"]'
+    link = context.browser.find_by_xpath(xpath_string)
+    test_helper.assertEqual(link["href"], f"{Config.RH_UI_URL}{expected_path}")
+
+
+# Cookies page radio interactions
+
+
+@step("the user sets the selection under {para_title} to {cookie_selection}")
+def change_cookies_selection(context, para_title, cookie_selection):
+    match para_title:
+        case "Cookies that measure website use":
+            name_attribute_value = "cookies-usage"
+        case "Cookies that help with our communications":
+            name_attribute_value = "cookies-campaigns"
+        case "Cookies that remember your settings":
+            name_attribute_value = "cookies-settings"
+
+    xpath_string = f'//input[@name="{name_attribute_value}" and @value="{cookie_selection.lower()}"]'
+    context.browser.find_by_xpath(xpath_string).first.click()
+
+    submit_button_xpath_string = '//button[@type="submit"]'
+    context.browser.find_by_xpath(submit_button_xpath_string).click()
+
+
+# Checking cookie values
+
+
+@step(
+    "the field {cookie_key} within the ons_cookie_policy cookie is set to {cookie_selection}"
+)
+def assert_ons_cookie_policy(context, cookie_key, cookie_selection):
+    ons_cookie_policy_dict = _parse_ons_cookie_policy_string(context)
+    actual_value = ons_cookie_policy_dict[cookie_key]
+    expected_value = True if cookie_selection == "On" else False
```

Comment:
Could perhaps use `strotobool` here instead so that other "truthy" values wouldn't become `False`.

Reply:
Well spotted, that's a shame as it's a handy little utility we'll probably need to replicate ourselves in a bunch of other places now.

I wouldn't be bothered about this one bit of code, but since we already used `strtobool` in a couple of places in the ATs, it's perhaps worth replacing it with our own implementation now? I won't block on it though, it's not critical so I don't mind if you'd rather offload it into a card on it's own.

Reply:
This will be picked up when migrating to Python 3.12, so I don't think we have to rush with it now - I think we've spent enough time on this card 😄

I've added [SDCSRM-725 Migrate to Python 3.12](https://jira.ons.gov.uk/browse/SDCSRM-725) as a low-priority Tech Session card and added some details about it.